c/indexer.c

```diff
@@ -90,8 +108,8 @@ static struct gzip_index *addpoint(struct gzip_index *index, uint8_t bits,
     /* fill in entry and increment how many we have */
     next = index->list + index->have;
     next->bits = bits;
-    next->in = in;
-    next->out = out;
+    next->in = encode_offset(in);
+    next->out = encode_offset(out);
     if (left)
         memcpy(next->window, window + WINSIZE - left, left);
```

Comment:
Since the element type for the dictionary is the unsigned char, which size is 1 byte precisely, it does not need to be encoded. BE / LE placement goes by bytes.